blueprint/core/blueprint_AppHarness.cpp

```diff
@@ -30,7 +30,23 @@ namespace blueprint
             for (const auto& f : fileWatcher->getWatchedFiles())
             {
                 if (onBeforeEach) { onBeforeEach(f); }
-                appRoot.evaluate(f);
+
+                try
+                {
+                    appRoot.evaluate(f);
+                }
+                catch (const EcmascriptEngine::Error& err)
+                {
+                    // We may fall in here in Release builds when ReactApplicationRoot hasn't
+                    // caught the error for us to show the red screen. In this case, we call
+                    // a user supplied error handler if we have one, else just break the loop
+                    // and await the next file change event.
+                    if (onEvalError)
+                        onEvalError(err);
```

Comment:
We could consider re-throwing here if no `onEvalError` callback supplied otherwise we're failing silently which feels a bit odd. That being said I'd never personally have the file watch timer thread enabled in Release myself so I wouldn't hit this codepath.

Reply:
Updated!